public/javascripts/context_modules.js

```diff
@@ -203,40 +229,74 @@ import 'compiled/jquery.rails_flash_notifications'
         })
       },
 
-      updateCourseProgress: function(callback){
-
-        function calculateUnitProgress(item){
-          if(item.items_count){
-            let total_items = 0;
-            var total_completed = [];
+      calculateUnitProgress: function (item){
+        if(item.items_count){
+          let total_items = 0;
+          var total_completed = [];
 
-            item.items.map(function(module){
-              if (module.completion_requirement) {
-                total_items++;
-              }
-              let cr = module.completion_requirement || {};
-              if(cr.completed){
-                total_completed.push(module);
-              }
-            });
-            if (total_items) {
-              return Math.floor((total_completed.length / total_items) * 100);
+          item.items.map(function(module){
+            if (module.completion_requirement) {
+              total_items++;
             }
+            let cr = module.completion_requirement || {};
+            if(cr.completed){
+              total_completed.push(module);
+            }
+          });
+          if (total_items) {
+            return Math.floor((total_completed.length / total_items) * 100);
           }
-          return 0;
         }
+        return 0;
+      },
+
+      updateCourseProgress: function(callback){
+
+        let $contextModules = $("#context_modules .context_module");
 
         var courseItems = JSON.parse(window.localStorage.getItem("course_items"));
 
-        courseItems.map(function(item){
-          let progressBar = $("#"+item.id+" .sm-unit-header_progress-container .sm-unit-header_progress-bar");
-          let progressText = $("#"+item.id+" .sm-unit-header_progress-container .sm-progress-percentage");
-          let progress = calculateUnitProgress(item);
+        courseItems.map((unit) => {
+          let progressBar = $("#"+unit.id+" .sm-unit-header_progress-container .sm-unit-header_progress-bar");
+          let progressText = $("#"+unit.id+" .sm-unit-header_progress-container .sm-progress-percentage");
+          let progress = modules.calculateUnitProgress(unit);
```

Comment:
was there a hick up with unit progress?"

Reply:
No. Renamed a variable and incorporated 2 loops over the same dom elements into 1 loop.

Reply:
Right. I was attempting to understand why. It looks like a minor refactor.

Reply:
Beautiful!